src/qibo/validation/gate_set_tomography.py

```diff
@@ -0,0 +1,2089 @@
+        if k == 0:  # |0>
+            circ = Circuit(nqubits, density_matrix=True)
+
+        elif k == 1:  # |1>
+            circ = Circuit(nqubits, density_matrix=True)
+            circ.add(gates.X(0))
+
+        elif k == 2:  # |+>
+            circ = Circuit(nqubits, density_matrix=True)
+            circ.add(gates.H(0))
+
+        elif k == 3:  # |y+>
+            circ = Circuit(nqubits, density_matrix=True)
+            circ.add(gates.H(0))
+            circ.add(gates.S(0))
+
+        # If no class_qubit_gate
+        if class_qubit_gate is None:
+            pass
+
+        # If class_qubit_gate is present (with/without theta)
+        elif class_qubit_gate is not None:
+            if theta is None:
+                print(f"Gate set tomography for 1 qubit gate: {class_qubit_gate}.")
+                circ.add(getattr(gates, class_qubit_gate)(0))
+            elif theta is not None:
+                print(
+                    f"Gate set tomography for 1 qubit gate: {class_qubit_gate}({theta})."
+                )
+                circ.add(getattr(gates, class_qubit_gate)(0, theta))
+
+        # ================= START OF 4 MEASUREMENT BASES =================
+        for j in range(0, 4):
+            if j == 0:  # Identity basis
+                circ2 = Circuit(nqubits, density_matrix=True)
+                new_circ = circ + circ2
+                if noise_model is not None and backend.name != "qibolab":
+                    new_circ = noise_model.apply(new_circ)
+                new_circ.add(gates.M(0, basis=gates.Z))
+                expectation_val = GST_measure_1qb(new_circ, k, j)
+
+            elif j == 1:  # X basis
+                circ2 = Circuit(nqubits, density_matrix=True)
+                new_circ = circ + circ2
+                if noise_model is not None and backend.name != "qibolab":
+                    new_circ = noise_model.apply(new_circ)
+                new_circ.add(gates.M(0, basis=gates.X))
+                expectation_val = GST_measure_1qb(new_circ, k, j)
+
+            elif j == 2:  # Y basis
+                circ2 = Circuit(nqubits, density_matrix=True)
+                new_circ = circ + circ2
+                if noise_model is not None and backend.name != "qibolab":
+                    new_circ = noise_model.apply(new_circ)
+                new_circ.add(gates.M(0, basis=gates.Y))
+                expectation_val = GST_measure_1qb(new_circ, k, j)
+
+            elif j == 3:  # Z basis
+                circ2 = Circuit(nqubits, density_matrix=True)
+                new_circ = circ + circ2
+                if noise_model is not None and backend.name != "qibolab":
+                    new_circ = noise_model.apply(new_circ)
+                new_circ.add(gates.M(0, basis=gates.Z))
+                expectation_val = GST_measure_1qb(new_circ, k, j)
+
+            matrix_jk_1qb[j, k] = expectation_val
+
+    matrix_jk_1qb = np.matrix(matrix_jk_1qb)
+
+    return matrix_jk_1qb
+
+
+def GST_2qb(
```

Comment:
Many of the comments that I made above for the 1 qubit GST apply here as well.

Reply:
Thanks Andrea. I've amended them.